packages/react-components/src/templates/__tests__/Layout.test.tsx

```diff
@@ -86,3 +88,16 @@ it('closes the drawer on navigation', async () => {
   userEvent.click(getAllByText(/network/i, { selector: 'nav *' })[0]);
   expect(getByLabelText(/close/i)).not.toBeVisible();
 });
+
+it('Scrolls to top between page navigations', async () => {
+  const scrollToMock = jest.fn();
+  Element.prototype.scrollTo = scrollToMock;
+
+  const { getAllByText } = render(
+    <MemoryRouter initialEntries={['/']}>
+      <Layout {...props} />
+    </MemoryRouter>,
+  );
+  userEvent.click(getAllByText(/network/i, { selector: 'nav *' })[0]);
+  expect(scrollToMock).toHaveBeenCalled();
```

Comment:
needs to check which element was scrolled (put some text children, intercept the `this`, and check that it has the right text content)